MREUnityRuntime/MREUnityRuntimeLib/Core/PhysicsBridge.cs

```diff
@@ -208,8 +172,13 @@ public void FixedUpdate(UnityEngine.Transform rootTransform)
 					continue;
 				}
 
+				// <todo> make sure that this will always hit this loop ???
 				// Find corresponding rigid body info.
-				while (index < snapshot.RigidBodies.Count && rb.Id.CompareTo(snapshot.RigidBodies.Values[index].Id) > 0) index++;
+				index = 0;
```

Comment:
RBs should be sorted for both arrays, we should not return index to 0 in every iteration

Reply:
yap.